### PR TITLE
[Stats Refresh] Add Period > Published card

### LINKED ISSUE #10902: [Stats Refresh] Add Period > Published card
- [ ] Published card showing Posts published within selected timeframe on DWMY views
- [ ] When row is tapped open post

(8)

Design:
published

### PR DESCRIPTION
**Pull request by ScoutHarris**
Fixes #10902 

This add the Period Published card.
- Max 6 data rows.
- If more than 6, `View more` is added.
- If no data rows, `No data for this period` is displayed.
- Data rows:
  - No icon is displayed.
  - Selecting a row will display the post in a web view.

To test:

---
- On a site with published content, go to Stats > select a period.
- Verify the view is:
published

---
- Select a row.
- Verify the content is displayed in a web view.

---
- On a site with _no_ published content, go to Stats > select a period.
- Verify the view is:
no_published

### CODE REVIEW

**Comment by ScoutHarris** [comment]
@SylvesterWilmott - 

published

---
no_published

---
ipad_portrait

---
ipad_landscape

**Comment by SylvesterWilmott** [comment]
Looks good! Thanks @ScoutHarris

File: WordPress/Classes/ViewRelated/Stats/Period Stats/SiteStatsPeriodViewModel.swift [hunk 1]
```diff
@@ -104,4 +106,29 @@ private extension SiteStatsPeriodViewModel {
...
+    func publishedTableRows() -> [ImmuTableRow] {
+        var tableRows = [ImmuTableRow]()
+        tableRows.append(CellHeaderRow(title: PeriodHeaders.published))
+        tableRows.append(TopTotalsNoSubtitlesPeriodStatsRow(dataRows: publishedDataRows(),
+                                                            siteStatsPeriodDelegate: periodDelegate))
+
+        return tableRows
+    }
+
+    func publishedDataRows() -> [StatsTotalRowData] {
```
Comment (jklausa) [inline WordPress/Classes/ViewRelated/Stats/Period Stats/SiteStatsPeriodViewModel.swift]: Same thing as in #10895 — this could be a simple `map` :)

Comment (ScoutHarris) [reply WordPress/Classes/ViewRelated/Stats/Period Stats/SiteStatsPeriodViewModel.swift]: And updated. Thanks!!

**Pull request merged**